### PR TITLE
OWLS-94850 - decouple logstash config from Operator

### LINKED ISSUE #104: Logstash config map
Hello,
currently logstash can be configured just rebuilding the operator image everytime.

I think that it would be nice to decouple the two things and have a config map to setup the logstash.conf.

The user could want to add different filters, save the logs also to file or send to a private ES instance and so on. 

For this you could provide a template with parametrized basic configuration. In the input parameter file let the users decide if deploy the whole ES stack or just logstash.

### PR DESCRIPTION
**Pull request by alai8**
Fixes #104

- Updated helm chart templates to put logstash.conf into new weblogic-operator-logstash-cm configmap
- Added new variable createLogStashConfigMap to optionally opt out creating weblogic-operator-logstash-cm config map during helm install
- Updated documentation related to logstash in operator

jenkins: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8215/

jenkins run on main for comparison: https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8216/

- additional failures in run 8215:
oracle.weblogic.kubernetes.ItKubernetesEvents.testScaleDomainAndVerifyCompletedEvent
- pass in re-run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8237/

oracle.weblogic.kubernetes.ItIstioMonitoringExporter.testIstioPrometheusViaExporterWebApp
oracle.weblogic.kubernetes.ItIstioMonitoringExporter.testIstioPrometheusWithSideCar
- pass in re-run https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/8229/

### CODE REVIEW

File: documentation/staging/content/userguide/managing-operators/using-helm.md [hunk 1]
```diff
@@ -504,6 +505,20 @@ Example:
 elasticSearchPort: 9201
 ```
 
+##### `createLogStashConfigMap`
+Specifies whether a ConfigMap named `weblogic-operator-logstash-cm` should be created during `helm install`. 
+The ConfigMap contains Logstash pipeline configuration for the Logstash container running in the operator pod.
```
Comment (rosemarymarano) [inline documentation/staging/content/userguide/managing-operators/using-helm.md:510]: contains Logstash pipeline configuration -> contains the Logstash pipeline configuration

File: documentation/staging/content/userguide/managing-operators/using-helm.md [hunk 2]
```diff
@@ -504,6 +505,20 @@ Example:
...
+If set to `true`, a ConfigMap will be created during `helm install` using the `logstash.conf` file in `kubernetes/samples/charts/weblogic-operator` directory.
```
Comment (rosemarymarano) [inline documentation/staging/content/userguide/managing-operators/using-helm.md:511]: in `kubernetes/samples/charts/weblogic-operator` directory. -> in the `kubernetes/samples/charts/weblogic-operator` directory.

File: documentation/staging/content/userguide/managing-operators/using-helm.md [hunk 3]
```diff
@@ -504,6 +505,20 @@ Example:
...
+Set `createLogStashConfigMap` to `false` if the ConfigMap already exists in the operator's namespace with Logstash configuration provided by the customer.
```
Comment (rosemarymarano) [inline documentation/staging/content/userguide/managing-operators/using-helm.md:512]: with Logstash configuration provided by the customer. -> with the Logstash configuration provided by the user.

**Review by rosemarymarano — COMMENTED** [review]
Minor edits. One comment on the architecture image. All the space labels are lowercase, except Load balancer -> load balancer.

**Comment by alai8** [comment]
Thanks for the comments. Minor edits and architecture diagram fixed in https://github.com/oracle/weblogic-kubernetes-operator/pull/2736/commits/9165a263daf75ef24a4304f2a6a7003d7252694f

**Review by rosemarymarano — APPROVED** [review]

**Review by rjeberhard — APPROVED** [review]

**Pull request merged**